### ISSUE TITLE
Snake Case Bug 🐛 : Run-time exception (line x): Index was outside the bounds of the array.

### ISSUE THREAD

**Issue by keizrivas** [SzymonHalucha/Minerals.StringCases#1]
**How to reproduce it:** 

In the following example, everything works as expected:

```
var str = "TestAbc";
Console.WriteLine(str.ToSnakeCase());

// Output: test_abc
```

But if you try with "TestA", "TestAb" or "TestABCDE" you get an exception:

```
var str = "TestAb";
Console.WriteLine(str.ToSnakeCase());

// Output: Run-time exception (line X): Index was outside the bounds of the array.
```

> Run-time exception (line X): Index was outside the bounds of the array.
>
> Stack Trace:
> [System.IndexOutOfRangeException: Index was outside the bounds of the array.]
>    at Minerals.StringCases.StringExtensions.ToSnakeCase(String value)
>    at Program.Main() :line X

**Comment by SzymonHalucha** [comment]
Hello Carlos,

Thank you for reporting this bug. I fixed it, this error was occurring in all cases. I added your cases to test project.

The new version (0.2.0) of the package is now available on NuGet!

**Comment by keyzzrivas** [comment]
Hi Szymon,

Thank you very much for your quick response!

**Comment by keizrivas** [comment]
Hi Szymon,

Thank you very much for your quick response!

**Comment by SzymonHalucha** [comment]
I was a bit too quick :) The bug has been completely fixed in version 0.2.1.
